source/Calamari/TerraformCliExecutor.cs

```diff
@@ -193,29 +194,49 @@ void InitializePlugins()
             var allowPluginDownloads = variables.GetFlag(TerraformSpecialVariables.Action.Terraform.AllowPluginDownloads, true);
             string initCommand = $"init -no-color";
 
-            if (version.IsLessThan("0.15.0"))
+            if (version?.IsLessThan("0.15.0") == true)
                 initCommand += $" -get-plugins={allowPluginDownloads.ToString().ToLower()}";
 
             initCommand += $" {initParams}";
 
             ExecuteCommandAndVerifySuccess(new[] { initCommand }, out _, true);
         }
 
-        Version GetVersion()
+        class TerraformVersionCommandOutput
         {
-            ExecuteCommandAndVerifySuccess(new[] { "--version" }, out string consoleOutput, true);
+            [JsonProperty("terraform_version")]
+            public string Version { get; set; }
+        }
 
-            consoleOutput = consoleOutput.Replace("Terraform v", "");
-            int newLinePos = consoleOutput.IndexOf('\n'); // this is always \n in all OSes for unknown reasons
-            if (newLinePos != -1)
-                consoleOutput = consoleOutput.Substring(0, newLinePos);
+        Version GetVersion()
+        {
+            ExecuteCommandAndVerifySuccess(new[] { "version --json" }, out string consoleOutput, true);
 
-            if (!Version.TryParse(consoleOutput, out var version))
+            Version parsedVersion = null;
+            bool hasParsingFailed = false;
+            var versionJsonOutput = JsonConvert.DeserializeObject<TerraformVersionCommandOutput>(consoleOutput,  new JsonSerializerSettings
             {
-                log.Warn($"Could not determine Terraform CLI version.");
+                // this prevents NewtonsoftJson from throwing an exception
+                Error = delegate(object sender, ErrorEventArgs args)
+                        {
+                            hasParsingFailed = true;
+                            args.ErrorContext.Handled = true;
+                        }
+            });
+            
+            if (hasParsingFailed || !Version.TryParse(versionJsonOutput.Version, out parsedVersion))
+            {
+                // fallback to regex match for older versions
+                var versionString = Regex.Match(consoleOutput, @"Terraform v([0-9\.]*)");
```

Comment:
I assume it will just ignore the `--json` switch in older versions then (i.e. we don't have to issue another command)

Reply:
Yes, it outputs the same as if `--json` isn't there.